pkg/cluster/cluster.go

```diff
@@ -1733,16 +1733,56 @@ func (c *Cluster) GetStatus() *ClusterStatus {
 
 // Switchover does a switchover (via Patroni) to a candidate pod
 func (c *Cluster) Switchover(curMaster *v1.Pod, candidate spec.NamespacedName) error {
-
 	var err error
 	c.logger.Debugf("switching over from %q to %q", curMaster.Name, candidate)
+
+	if !isInMaintenanceWindow(c.Spec.MaintenanceWindows) {
+		c.logger.Infof("postponing switchover, not in maintenance window")
+
+		var possibleSwitchover, schedule time.Time
+
+		now := time.Now().UTC()
+		for _, window := range c.Spec.MaintenanceWindows {
+			if window.Everyday {
+				possibleSwitchover = time.Date(now.Year(), now.Month(), now.Day(), window.StartTime.Hour(), window.StartTime.Minute(), 0, 0, time.UTC)
+				if now.After(possibleSwitchover) {
```

Comment:
```suggestion
			possibleSwitchover = time.Date(now.Year(), now.Month(), now.Day(), window.StartTime.Hour(), window.StartTime.Minute(), 0, 0, time.UTC)
			if window.Everyday && now.After(possibleSwitchover) {
```

Reply:
this way "else" path's logic with Weekday attribute will be executed for "window is everyday but is after the current time.

Reply:
ok. but I started to run tests and now see the general problem here...